test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -301,6 +311,22 @@ TEST_P(WasmHttpFilterTest, Metadata) {
   filter_->log(&request_headers, nullptr, nullptr, log_stream_info);
 }
 
+// Null VM Plugin, headers only.
+TEST_F(WasmHttpFilterTest, NullVmPluginRequestHeadersOnly) {
```

Comment:
This crashes under ASan:
```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==38203==ERROR: AddressSanitizer: SEGV on unknown address 0x000000023d50 (pc 0x7f535c775f8b bp 0x7ffda91e7d10 sp 0x7ffda91e74b8 T0)
==38203==The signal is caused by a WRITE memory access.
    #0 0x7f535c775f8a  (/lib/x86_64-linux-gnu/libc.so.6+0x83f8a)
    #1 0x58b4a5 in __asan_memcpy /build/llvm-toolchain-8-8.0.1~svn357234/projects/compiler-rt/lib/asan/asan_interceptors_memintrinsics.cc:23:3
    #2 0x7f537267c84b in Envoy::Extensions::Common::Wasm::Wasm::copyToPointerSize(absl::string_view, unsigned long, unsigned long) /proc/self/cwd/bazel-out/k8-opt/bin/source/extensions/common/wasm/_virtual_includes/wasm_hdr/extensions/common/wasm/wasm.h:860:5
    #3 0x7f53726808f9 in Envoy::Extensions::Common::Wasm::getHeaderMapValueHandler(void*, Envoy::Extensions::Common::Wasm::Word, Envoy::Extensions::Common::Wasm::Word, Envoy::Extensions::Common::Wasm::Word, Envoy::Extensions::Common::Wasm::Word, Envoy::Extensions::Common::Wasm::Word) /proc/self/cwd/source/extensions/common/wasm/wasm.cc:356:20
    #4 0x7f5372f6ea53 in Envoy::Extensions::Common::Wasm::Null::Plugin::proxy_getHeaderMapValue(Envoy::Extensions::Common::Wasm::HeaderMapType, char const*, unsigned long, char const**, unsigned long*) /proc/self/cwd/bazel-out/k8-opt/bin/source/extensions/common/wasm/null/_virtual_includes/null_plugin_hdr/extensions/common/wasm/null/proxy_wasm_externs.h:85:3
    #5 0x7f5372f6ea53 in Envoy::Extensions::Common::Wasm::Null::Plugin::getHeaderMapValue(Envoy::Extensions::Common::Wasm::HeaderMapType, absl::string_view) /proc/self/cwd/./api/wasm/cpp/proxy_wasm_impl.h:598
    #6 0x7f5372f6ea53 in Envoy::Extensions::Common::Wasm::Null::Plugin::getRequestHeader(absl::string_view) /proc/self/cwd/./api/wasm/cpp/proxy_wasm_impl.h:628
    #7 0x7f5372f6ea53 in Envoy::Extensions::Common::Wasm::Null::Plugin::ExamplePlugin::PluginContext::onRequestHeaders() /proc/self/cwd/source/extensions/common/wasm/null/plugin.cc:35
    #8 0x7f5372463bae in Envoy::Extensions::Common::Wasm::Null::NullVmPlugin::onRequestHeaders(unsigned long) /proc/self/cwd/source/extensions/common/wasm/null/null.cc:395:35
    #9 0x7f5372463bae in Envoy::Extensions::Common::Wasm::Null::NullVm::getFunction(absl::string_view, std::function<Envoy::Extensions::Common::Wasm::Word (Envoy::Extensions::Common::Wasm::Context*, Envoy::Extensions::Common::Wasm::Word)>*)::$_15::operator()(Envoy::Extensions::Common::Wasm::Context*, Envoy::Extensions::Common::Wasm::Word) const /proc/self/cwd/source/extensions/common/wasm/null/null.cc:306
    #10 0x7f5372463bae in std::_Function_handler<Envoy::Extensions::Common::Wasm::Word (Envoy::Extensions::Common::Wasm::Context*, Envoy::Extensions::Common::Wasm::Word), Envoy::Extensions::Common::Wasm::Null::NullVm::getFunction(absl::string_view, std::function<Envoy::Extensions::Common::Wasm::Word (Envoy::Extensions::Common::Wasm::Context*, Envoy::Extensions::Common::Wasm::Word)>*)::$_15>::_M_invoke(std::_Any_data const&, Envoy::Extensions::Common::Wasm::Context*&&, Envoy::Extensions::Common::Wasm::Word&&) /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/functional:1716
    #11 0x7f537269be6e in std::function<Envoy::Extensions::Common::Wasm::Word (Envoy::Extensions::Common::Wasm::Context*, Envoy::Extensions::Common::Wasm::Word)>::operator()(Envoy::Extensions::Common::Wasm::Context*, Envoy::Extensions::Common::Wasm::Word) const /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/functional:2127:14
    #12 0x7f537269bb7e in Envoy::Extensions::Common::Wasm::Context::onRequestHeaders() /proc/self/cwd/source/extensions/common/wasm/wasm.cc:1081:7
    ...
```

Note that the lines might be slightly off, since I have cherry-picked https://github.com/envoyproxy/envoy-wasm/pull/1 in order to disable WAVM, since it's leaking memory.

Reply:
This is with v8 then ? or the Null plugin?

Reply:
Ah, the test... I see.

Reply:
Found it.  another uint32_t hanging around.  Fixed.